aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -264,15 +266,24 @@ private boolean isGlueIcebergTable(Table table) {
 
   @Override
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    TableMetadata lastMetadata;
+    try {
+      lastMetadata = ops.current();
+    } catch (NoSuchKeyException ex) {
+      // S3 files already deleted, can not get S3 object through metadata location
+      lastMetadata = null;
+    }
     try {
```

Comment:
try should be at the top including `TableOperations ops = newTableOps(identifier);`